--- ../src-base/minecraft/net/minecraft/entity/player/InventoryPlayer.java
+++ ../src-work/minecraft/net/minecraft/entity/player/InventoryPlayer.java
@@ -32,6 +32,7 @@
     public final NonNullList<ItemStack> field_70462_a = NonNullList.<ItemStack>func_191197_a(36, ItemStack.field_190927_a);
     public final NonNullList<ItemStack> field_70460_b = NonNullList.<ItemStack>func_191197_a(4, ItemStack.field_190927_a);
     public final NonNullList<ItemStack> field_184439_c = NonNullList.<ItemStack>func_191197_a(1, ItemStack.field_190927_a);
+    public final NonNullList<ItemStack> capeInventory = NonNullList.<ItemStack>func_191197_a(1, ItemStack.field_190927_a);
     private final List<NonNullList<ItemStack>> field_184440_g;
     public int field_70461_c;
     public EntityPlayer field_70458_d;
@@ -40,7 +41,7 @@
 
     public InventoryPlayer(EntityPlayer p_i1750_1_)
     {
-        this.field_184440_g = Arrays.<NonNullList<ItemStack>>asList(this.field_70462_a, this.field_70460_b, this.field_184439_c);
+        this.field_184440_g = Arrays.<NonNullList<ItemStack>>asList(this.field_70462_a, this.field_70460_b, this.field_184439_c, this.capeInventory);
         this.field_70457_g = ItemStack.field_190927_a;
         this.field_70458_d = p_i1750_1_;
     }
@@ -292,7 +293,8 @@
 
         if (itemstack.func_190926_b())
         {
-            itemstack = new ItemStack(item, 0, p_191973_2_.func_77960_j());
+            itemstack = p_191973_2_.func_77946_l(); // Forge: Replace Item clone above to preserve item capabilities when picking the item up.
+            itemstack.func_190920_e(0);
 
             if (p_191973_2_.func_77942_o())
             {
@@ -363,6 +365,13 @@
                 }
             }
         }
+        for (ItemStack is : field_70460_b) // FORGE: Tick armor on animation ticks
+        {
+            if (!is.func_190926_b())
+            {
+                is.func_77973_b().onArmorTick(field_70458_d.field_70170_p, field_70458_d, is);
+            }
+        }
     }
 
     public boolean func_70441_a(ItemStack p_70441_1_)
@@ -613,6 +622,17 @@
             }
         }
 
+        for (int l = 0; l < this.capeInventory.size(); ++l)
+        {
+            if (!((ItemStack)this.capeInventory.get(l)).func_190926_b())
+            {
+                NBTTagCompound nbttagcompound3 = new NBTTagCompound();
+                nbttagcompound3.func_74774_a("Slot", (byte)(l + 110));
+                ((ItemStack)this.capeInventory.get(l)).func_77955_b(nbttagcompound3);
+                p_70442_1_.func_74742_a(nbttagcompound3);
+            }
+        }
+
         return p_70442_1_;
     }
 
@@ -621,6 +641,7 @@
         this.field_70462_a.clear();
         this.field_70460_b.clear();
         this.field_184439_c.clear();
+        this.capeInventory.clear();
 
         for (int i = 0; i < p_70443_1_.func_74745_c(); ++i)
         {
@@ -642,13 +663,17 @@
                 {
                     this.field_184439_c.set(j - 150, itemstack);
                 }
+                else if (j >= 110 && j < this.capeInventory.size() + 110)
+                {
+                    this.capeInventory.set(j - 110, itemstack);
+                }
             }
         }
     }
 
     public int func_70302_i_()
     {
-        return this.field_70462_a.size() + this.field_70460_b.size() + this.field_184439_c.size();
+        return this.field_70462_a.size() + this.field_70460_b.size() + this.field_184439_c.size() + this.capeInventory.size();
     }
 
     public boolean func_191420_l()
@@ -677,6 +702,14 @@
             }
         }
 
+        for (ItemStack itemstack2 : this.capeInventory)
+        {
+            if (!itemstack2.func_190926_b())
+            {
+                return false;
+            }
+        }
+
         return true;
     }
 
