--- ../src-base/minecraft/net/minecraft/network/EnumConnectionState.java
+++ ../src-work/minecraft/net/minecraft/network/EnumConnectionState.java
@@ -248,6 +248,7 @@
             this.func_179245_a(EnumPacketDirection.SERVERBOUND, CPacketSpectate.class);
             this.func_179245_a(EnumPacketDirection.SERVERBOUND, CPacketPlayerTryUseItemOnBlock.class);
             this.func_179245_a(EnumPacketDirection.SERVERBOUND, CPacketPlayerTryUseItem.class);
+            this.func_179245_a(EnumPacketDirection.SERVERBOUND, net.pl3x.forge.packet.CPacketPlayerLeftClickItem.class);
         }
     },
     STATUS(1)
