--- ../src-base/minecraft/net/minecraft/tileentity/TileEntitySign.java
+++ ../src-work/minecraft/net/minecraft/tileentity/TileEntitySign.java
@@ -59,7 +59,7 @@
             }
             public boolean func_70003_b(int p_70003_1_, String p_70003_2_)
             {
-                return true;
+                return p_70003_1_ <= 2; //Forge: Fixes  MC-75630 - Exploit with signs and command blocks
             }
             public BlockPos func_180425_c()
             {
@@ -141,70 +141,70 @@
 
     public boolean func_174882_b(final EntityPlayer p_174882_1_)
     {
-        ICommandSender icommandsender = new ICommandSender()
+        /*ICommandSender icommandsender = new ICommandSender()
         {
-            public String func_70005_c_()
+            public String getName()
             {
-                return p_174882_1_.func_70005_c_();
+                return playerIn.getName();
             }
-            public ITextComponent func_145748_c_()
+            public ITextComponent getDisplayName()
             {
-                return p_174882_1_.func_145748_c_();
+                return playerIn.getDisplayName();
             }
-            public void func_145747_a(ITextComponent p_145747_1_)
+            public void sendMessage(ITextComponent component)
             {
             }
-            public boolean func_70003_b(int p_70003_1_, String p_70003_2_)
+            public boolean canUseCommand(int permLevel, String commandName)
             {
-                return p_70003_1_ <= 2;
+                return permLevel <= 2;
             }
-            public BlockPos func_180425_c()
+            public BlockPos getPosition()
             {
-                return TileEntitySign.this.field_174879_c;
+                return TileEntitySign.this.pos;
             }
-            public Vec3d func_174791_d()
+            public Vec3d getPositionVector()
             {
-                return new Vec3d((double)TileEntitySign.this.field_174879_c.func_177958_n() + 0.5D, (double)TileEntitySign.this.field_174879_c.func_177956_o() + 0.5D, (double)TileEntitySign.this.field_174879_c.func_177952_p() + 0.5D);
+                return new Vec3d((double)TileEntitySign.this.pos.getX() + 0.5D, (double)TileEntitySign.this.pos.getY() + 0.5D, (double)TileEntitySign.this.pos.getZ() + 0.5D);
             }
-            public World func_130014_f_()
+            public World getEntityWorld()
             {
-                return p_174882_1_.func_130014_f_();
+                return playerIn.getEntityWorld();
             }
-            public Entity func_174793_f()
+            public Entity getCommandSenderEntity()
             {
-                return p_174882_1_;
+                return playerIn;
             }
-            public boolean func_174792_t_()
+            public boolean sendCommandFeedback()
             {
                 return false;
             }
-            public void func_174794_a(CommandResultStats.Type p_174794_1_, int p_174794_2_)
+            public void setCommandStat(CommandResultStats.Type type, int amount)
             {
-                if (TileEntitySign.this.field_145850_b != null && !TileEntitySign.this.field_145850_b.field_72995_K)
+                if (TileEntitySign.this.world != null && !TileEntitySign.this.world.isRemote)
                 {
-                    TileEntitySign.this.field_174883_i.func_184932_a(TileEntitySign.this.field_145850_b.func_73046_m(), this, p_174794_1_, p_174794_2_);
+                    TileEntitySign.this.stats.setCommandStatForSender(TileEntitySign.this.world.getMinecraftServer(), this, type, amount);
                 }
             }
-            public MinecraftServer func_184102_h()
+            public MinecraftServer getServer()
             {
-                return p_174882_1_.func_184102_h();
+                return playerIn.getServer();
             }
         };
 
-        for (ITextComponent itextcomponent : this.field_145915_a)
+        for (ITextComponent itextcomponent : this.signText)
         {
-            Style style = itextcomponent == null ? null : itextcomponent.func_150256_b();
+            Style style = itextcomponent == null ? null : itextcomponent.getStyle();
 
-            if (style != null && style.func_150235_h() != null)
+            if (style != null && style.getClickEvent() != null)
             {
-                ClickEvent clickevent = style.func_150235_h();
+                ClickEvent clickevent = style.getClickEvent();
 
-                if (clickevent.func_150669_a() == ClickEvent.Action.RUN_COMMAND)
+                if (clickevent.getAction() == ClickEvent.Action.RUN_COMMAND)
                 {
-                    p_174882_1_.func_184102_h().func_71187_D().func_71556_a(icommandsender, clickevent.func_150668_b());
+                    playerIn.getServer().getCommandManager().executeCommand(icommandsender, clickevent.getValue());
                 }
             }
-        }
+        }*/
 
         return true;
     }
