--- ../src-base/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java
+++ ../src-work/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java
@@ -126,10 +126,16 @@
             }
         }
 
-        if (this.field_78779_k.func_77145_d() && !this.field_78776_a.field_71439_g.func_184614_ca().func_190926_b() && this.field_78776_a.field_71439_g.func_184614_ca().func_77973_b() instanceof ItemSword)
+        ItemStack stack = field_78776_a.field_71439_g.func_184614_ca();
+        if (!stack.func_190926_b() && stack.func_77973_b().onBlockStartBreak(stack, p_187103_1_, field_78776_a.field_71439_g))
         {
             return false;
         }
+
+        if (this.field_78779_k.func_77145_d() && !stack.func_190926_b() && !stack.func_77973_b().canDestroyBlockInCreative(field_78776_a.field_71441_e, p_187103_1_, stack, field_78776_a.field_71439_g))
+        {
+            return false;
+        }
         else
         {
             World world = this.field_78776_a.field_71441_e;
@@ -147,19 +153,13 @@
             else
             {
                 world.func_175718_b(2001, p_187103_1_, Block.func_176210_f(iblockstate));
-                block.func_176208_a(world, p_187103_1_, iblockstate, this.field_78776_a.field_71439_g);
-                boolean flag = world.func_180501_a(p_187103_1_, Blocks.field_150350_a.func_176223_P(), 11);
 
-                if (flag)
-                {
-                    block.func_176206_d(world, p_187103_1_, iblockstate);
-                }
-
                 this.field_178895_c = new BlockPos(this.field_178895_c.func_177958_n(), -1, this.field_178895_c.func_177952_p());
 
                 if (!this.field_78779_k.func_77145_d())
                 {
                     ItemStack itemstack1 = this.field_78776_a.field_71439_g.func_184614_ca();
+                    ItemStack copyBeforeUse = itemstack1.func_77946_l();
 
                     if (!itemstack1.func_190926_b())
                     {
@@ -167,11 +167,18 @@
 
                         if (itemstack1.func_190926_b())
                         {
+                            net.minecraftforge.event.ForgeEventFactory.onPlayerDestroyItem(this.field_78776_a.field_71439_g, copyBeforeUse, EnumHand.MAIN_HAND);
                             this.field_78776_a.field_71439_g.func_184611_a(EnumHand.MAIN_HAND, ItemStack.field_190927_a);
                         }
                     }
                 }
 
+                boolean flag = block.removedByPlayer(iblockstate, world, p_187103_1_, field_78776_a.field_71439_g, false);
+
+                if (flag)
+                {
+                    block.func_176206_d(world, p_187103_1_, iblockstate);
+                }
                 return flag;
             }
         }
@@ -212,6 +219,7 @@
             {
                 this.field_78776_a.func_193032_ao().func_193294_a(this.field_78776_a.field_71441_e, p_180511_1_, this.field_78776_a.field_71441_e.func_180495_p(p_180511_1_), 1.0F);
                 this.field_78774_b.func_147297_a(new CPacketPlayerDigging(CPacketPlayerDigging.Action.START_DESTROY_BLOCK, p_180511_1_, p_180511_2_));
+                if (!net.minecraftforge.common.ForgeHooks.onLeftClickBlock(this.field_78776_a.field_71439_g, p_180511_1_, p_180511_2_, net.minecraftforge.common.ForgeHooks.rayTraceEyeHitVec(this.field_78776_a.field_71439_g, func_78757_d() + 1)).isCanceled())
                 func_178891_a(this.field_78776_a, this, p_180511_1_, p_180511_2_);
                 this.field_78781_i = 5;
             }
@@ -221,6 +229,7 @@
                 {
                     this.field_78774_b.func_147297_a(new CPacketPlayerDigging(CPacketPlayerDigging.Action.ABORT_DESTROY_BLOCK, this.field_178895_c, p_180511_2_));
                 }
+                net.minecraftforge.event.entity.player.PlayerInteractEvent.LeftClickBlock event = net.minecraftforge.common.ForgeHooks.onLeftClickBlock(this.field_78776_a.field_71439_g, p_180511_1_, p_180511_2_, net.minecraftforge.common.ForgeHooks.rayTraceEyeHitVec(this.field_78776_a.field_71439_g, func_78757_d() + 1));
 
                 IBlockState iblockstate = this.field_78776_a.field_71441_e.func_180495_p(p_180511_1_);
                 this.field_78776_a.func_193032_ao().func_193294_a(this.field_78776_a.field_71441_e, p_180511_1_, iblockstate, 0.0F);
@@ -229,9 +238,11 @@
 
                 if (flag && this.field_78770_f == 0.0F)
                 {
+                    if (event.getUseBlock() != net.minecraftforge.fml.common.eventhandler.Event.Result.DENY)
                     iblockstate.func_177230_c().func_180649_a(this.field_78776_a.field_71441_e, p_180511_1_, this.field_78776_a.field_71439_g);
                 }
 
+                if (event.getUseItem() == net.minecraftforge.fml.common.eventhandler.Event.Result.DENY) return true;
                 if (flag && iblockstate.func_185903_a(this.field_78776_a.field_71439_g, this.field_78776_a.field_71439_g.field_70170_p, p_180511_1_) >= 1.0F)
                 {
                     this.func_187103_a(p_180511_1_);
@@ -297,7 +308,7 @@
 
                 if (this.field_78780_h % 4.0F == 0.0F)
                 {
-                    SoundType soundtype = block.func_185467_w();
+                    SoundType soundtype = block.getSoundType(iblockstate, field_78776_a.field_71441_e, p_180512_1_, field_78776_a.field_71439_g);
                     this.field_78776_a.func_147118_V().func_147682_a(new PositionedSoundRecord(soundtype.func_185846_f(), SoundCategory.NEUTRAL, (soundtype.func_185843_a() + 1.0F) / 8.0F, soundtype.func_185847_b() * 0.5F, p_180512_1_));
                 }
 
@@ -597,4 +608,38 @@
     {
         this.field_78774_b.func_147297_a(new CPacketCustomPayload("MC|PickItem", (new PacketBuffer(Unpooled.buffer())).func_150787_b(p_187100_1_)));
     }
+
+    public EnumActionResult processLeftClick(EntityPlayer player, World worldIn, EnumHand hand)
+    {
+        if (this.field_78779_k == GameType.SPECTATOR)
+        {
+            return EnumActionResult.PASS;
+        }
+
+        this.func_78750_j();
+        this.field_78774_b.func_147297_a(new net.pl3x.forge.packet.CPacketPlayerLeftClickItem(hand));
+        ItemStack itemstack = player.func_184586_b(hand);
+
+        if (player.func_184811_cZ().func_185141_a(itemstack.func_77973_b()))
+        {
+            return EnumActionResult.PASS;
+        }
+
+        EnumActionResult cancelResult = net.minecraftforge.common.ForgeHooks.onItemLeftClick(player, hand);
+        if (cancelResult != null) return cancelResult;
+        int i = itemstack.func_190916_E();
+        ActionResult<ItemStack> actionresult = itemstack.onItemLeftClick(worldIn, player, hand);
+        ItemStack itemstack1 = actionresult.func_188398_b();
+
+        if (itemstack1 != itemstack || itemstack1.func_190916_E() != i)
+        {
+            player.func_184611_a(hand, itemstack1);
+            if (itemstack1.func_190926_b())
+            {
+                net.minecraftforge.event.ForgeEventFactory.onPlayerDestroyItem(player, itemstack, hand);
+            }
+        }
+
+        return actionresult.func_188397_a();
+    }
 }
