--- ../src-base/minecraft/net/minecraft/client/gui/GuiChat.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiChat.java
@@ -24,8 +24,8 @@
     private String field_146410_g = "";
     private int field_146416_h = -1;
     private TabCompleter field_184096_i;
-    protected GuiTextField field_146415_a;
-    private String field_146409_v = "";
+    public GuiTextField field_146415_a;
+    protected String field_146409_v = "";
 
     public GuiChat()
     {
