--- ../src-base/minecraft/net/minecraft/client/gui/GuiPlayerTabOverlay.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiPlayerTabOverlay.java
@@ -167,7 +167,7 @@
                 if (flag)
                 {
                     EntityPlayer entityplayer = this.field_175250_f.field_71441_e.func_152378_a(gameprofile.getId());
-                    boolean flag1 = entityplayer != null && entityplayer.func_175148_a(EnumPlayerModelParts.CAPE) && ("Dinnerbone".equals(gameprofile.getName()) || "Grumm".equals(gameprofile.getName()));
+                    boolean flag1 = entityplayer != null && entityplayer.func_175148_a(EnumPlayerModelParts.CAPE);
                     this.field_175250_f.func_110434_K().func_110577_a(networkplayerinfo1.func_178837_g());
                     int l2 = 8 + (flag1 ? 8 : 0);
                     int i3 = 8 * (flag1 ? -1 : 1);
