--- ../src-base/minecraft/net/minecraft/client/gui/GuiSleepMP.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiSleepMP.java
@@ -32,11 +32,11 @@
 
             if (!s.isEmpty())
             {
-                this.field_146297_k.field_71439_g.func_71165_d(s);
+                this.func_175275_f(s); // Forge: fix vanilla not adding messages to the sent list while sleeping
             }
 
             this.field_146415_a.func_146180_a("");
-            this.field_146297_k.field_71456_v.func_146158_b().func_146240_d();
+            this.field_146297_k.field_71456_v.getChatGUI().resetScroll();
         }
     }
 
