--- ../src-base/minecraft/net/minecraft/client/gui/GuiScreenBook.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiScreenBook.java
@@ -38,14 +38,14 @@
     private final ItemStack field_146474_h;
     private final boolean field_146475_i;
     private boolean field_146481_r;
-    private boolean field_146480_s;
+    public boolean field_146480_s;
     private int field_146479_t;
     private final int field_146478_u = 192;
     private final int field_146477_v = 192;
     private int field_146476_w = 1;
     private int field_146484_x;
     private NBTTagList field_146483_y;
-    private String field_146482_z = "";
+    public String field_146482_z = "";
     private List<ITextComponent> field_175386_A;
     private int field_175387_B = -1;
     private GuiScreenBook.NextPageButton field_146470_A;
@@ -69,6 +69,7 @@
 
             if (this.field_146476_w < 1)
             {
+                this.field_146483_y.func_74742_a(new NBTTagString("")); // Forge: fix MC-1685
                 this.field_146476_w = 1;
             }
         }
@@ -319,12 +320,12 @@
         }
     }
 
-    private String func_146456_p()
+    public String func_146456_p()
     {
         return this.field_146483_y != null && this.field_146484_x >= 0 && this.field_146484_x < this.field_146483_y.func_74745_c() ? this.field_146483_y.func_150307_f(this.field_146484_x) : "";
     }
 
-    private void func_146457_a(String p_146457_1_)
+    public void func_146457_a(String p_146457_1_)
     {
         if (this.field_146483_y != null && this.field_146484_x >= 0 && this.field_146484_x < this.field_146483_y.func_74745_c())
         {
