--- ../src-base/minecraft/net/minecraft/inventory/ContainerPlayer.java
+++ ../src-work/minecraft/net/minecraft/inventory/ContainerPlayer.java
@@ -43,7 +43,7 @@
                 }
                 public boolean func_75214_a(ItemStack p_75214_1_)
                 {
-                    return entityequipmentslot == EntityLiving.func_184640_d(p_75214_1_);
+                    return p_75214_1_.func_77973_b().isValidArmor(p_75214_1_, entityequipmentslot, field_82862_h);
                 }
                 public boolean func_82869_a(EntityPlayer p_82869_1_)
                 {
@@ -81,6 +81,28 @@
                 return "minecraft:items/empty_armor_slot_shield";
             }
         });
+        this.func_75146_a(new Slot(p_i1819_1_, 41, 77, 44)
+        {
+            @Nullable
+            @SideOnly(Side.CLIENT)
+            public String func_178171_c()
+            {
+                return "minecraft:items/empty_armor_slot_banner";
+            }
+
+            @Override
+            public boolean func_75214_a(ItemStack stack)
+            {
+                return stack.func_77973_b() == net.minecraft.init.Items.field_179564_cE;
+            }
+
+            @Override
+            public void func_75218_e()
+            {
+                net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.pl3x.forge.event.BannerSlotChangedEvent(ContainerPlayer.this, this));
+                super.func_75218_e();
+            }
+        });
     }
 
     public void func_75130_a(IInventory p_75130_1_)
