--- ../src-base/minecraft/net/minecraft/client/gui/chat/NormalChatListener.java
+++ ../src-work/minecraft/net/minecraft/client/gui/chat/NormalChatListener.java
@@ -18,6 +18,6 @@
 
     public void func_192576_a(ChatType p_192576_1_, ITextComponent p_192576_2_)
     {
-        this.field_192581_a.field_71456_v.func_146158_b().func_146227_a(p_192576_2_);
+        this.field_192581_a.field_71456_v.getChatGUI().printChatMessage(p_192576_2_);
     }
 }
